2000-11-04  Jakub Jelinek  <jakub@redhat.com>

	* function.c (expand_function_start): Cast GET_MODE_SIZE to
	HOST_WIDE_INT before negating it.

--- gcc/function.c.jj	Mon Oct 23 15:24:47 2000
+++ gcc/function.c	Sat Nov  4 08:42:20 2000
@@ -6200,7 +6200,8 @@ expand_function_start (subr, parms_have_
 	  /* Chain thru stack frames, assuming pointer to next lexical frame
 	     is found at the place we always store it.  */
 #ifdef FRAME_GROWS_DOWNWARD
-	  last_ptr = plus_constant (last_ptr, - GET_MODE_SIZE (Pmode));
+	  last_ptr = plus_constant (last_ptr,
+	  			    - (HOST_WIDE_INT) GET_MODE_SIZE (Pmode));
 #endif
 	  last_ptr = gen_rtx_MEM (Pmode, memory_address (Pmode, last_ptr));
 	  MEM_ALIAS_SET (last_ptr) = get_frame_alias_set ();
